instrumentation/opentelemetry-instrumentation-pika/src/opentelemetry/instrumentation/pika/pika_instrumentor.py

```diff
@@ -101,5 +94,24 @@ def _uninstrument(self, **kwargs: Dict[str, Any]) -> None:
                 channel._impl._consumers[key] = callback._original_callback
         PikaInstrumentor._uninstrument_channel_functions(channel)
 
+    def _decorate_channel_function(
+        self, tracer_provider: Optional[TracerProvider]
+    ) -> None:
+        self.original_channel_func = BlockingConnection.channel
+
+        def _wrapper(*args, **kwargs):
+            channel = self.original_channel_func(*args, **kwargs)
+            self.instrument_channel(channel, tracer_provider=tracer_provider)
```

Comment:
does this mean instrument_channel gets called on every single call to the underlying channel function?

Reply:
No, `_decorate_channel_function` monkey-patches the connection object. The `_wrapper` (and therefore the `instrument_channel`) is called only upon the creation of the channel.